x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/Realms.java

```diff
@@ -128,9 +113,8 @@ public Realms(Settings settings, Environment env, Map<String, Realm.Factory> fac
             return Collections.unmodifiableList(realms);
         }
 
-        AllowedRealmType allowedRealmType = licenseState.allowedRealmType();
         // If all realms are allowed, then nothing is unlicensed
-        if (allowedRealmType == AllowedRealmType.ALL) {
+        if (licenseState.areAllRealmsAllowed()) {
```

Comment:
For correctness, this method should also get a snapshot of the license state and possibly even pass it into an internal version of `asList()` that accepts the snapshot.